### ISSUE TITLE
In singularity exec, host environment variable overriding value set in docker image.

### ISSUE THREAD

**Issue by ilveroluca** [sylabs/singularity#533]
**Version of Singularity**
What version of Singularity are you using? Run:
```
$ singularity --version
singularity-ce version 3.9.2-focal
```

**Describe the bug**
In `singularity exec`, host environment variable overriding environment variable set in docker image.

**To Reproduce**
We can use the Docker openjdk image from dockerhub.  The image defines the environment variable `JAVA_HOME=/usr/local/openjdk-8`:
```
$ docker inspect -f '{{.Config.Env}}'  openjdk:8
[PATH=/usr/local/openjdk-8/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin JAVA_HOME=/usr/local/openjdk-8 LANG=C.UTF-8 JAVA_VERSION=8u312]
$ docker run --rm -it openjdk:8 bash -c 'echo $JAVA_HOME'
/usr/local/openjdk-8
```

Now let's run the same image with singularity.  If the host does not define JAVA_HOME, everything works as expected:
```
$ env | grep JAVA_HOME
$ singularity  exec docker://openjdk:8 bash -c 'echo $JAVA_HOME'
INFO:    Using cached SIF image
/usr/local/openjdk-8
```

On the other hand, if the host has a value set for JAVA_HOME:
```
$ JAVA_HOME=/simple/test/ singularity  exec docker://openjdk:8 bash -c 'echo $JAVA_HOME'
INFO:    Using cached SIF image
/simple/test/
```

**Expected behavior**
I expect the environment variable set by the container image to override the variable of the same name set on the host (I believe this is what [the documentation says](https://sylabs.io/guides/3.9/user-guide/environment_and_metadata.html#environment-from-the-host).  Therefore, I expect the value of JAVA_HOME set by the image to take precedence over the value set on the host.

**OS / Linux Distribution**
Which Linux distribution are you using?
```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Installation Method**
deb package from https://github.com/sylabs/singularity/releases

**Additional context**
None

**Comment by dtrudg** [comment]
Addressed in the docs, via https://github.com/sylabs/singularity-userdocs/issues/69

This is now live in the 3.9 docs. See edits at:

https://sylabs.io/guides/3.9/user-guide/environment_and_metadata.html#environment-from-a-base-image
https://sylabs.io/guides/3.9/user-guide/environment_and_metadata.html#environment-from-the-host
https://sylabs.io/guides/3.9/user-guide/singularity_and_docker.html#environment-propagation

(you may need to force refresh the page if it is cached in your browser)

**Comment by dtrudg** [comment]
Hi @ilveroluca - wanted to post a note here that it is possible portions of the docs updates are incorrect, and there is an underlying regression here that I am now investigating.

Apologies for the inconvenience.

**Comment by dtrudg** [comment]
Hi @ilveroluca - sorry about all the confusion here. This was confirmed as a regression in the end, introduced by a dependency update in 3.9.2 as detailed in #543 

The fix is released now in CE 3.9.5 which is available for download.

I'll go back to the documentation and re-apply my clarifications, but with a correction on the docker env var override (via SINGULARITYENV_ / --env / --env-file but not bare host env vars).

Thanks for your patience.
